native/explorer/src/series.rs

```diff
@@ -1231,65 +1231,55 @@ pub fn s_n_distinct(s: ExSeries) -> Result<usize, ExplorerError> {
 }
 
 #[rustler::nif(schedule = "DirtyCpu")]
-pub fn s_pow(s: ExSeries, other: ExSeries) -> Result<ExSeries, ExplorerError> {
-    match (s.dtype().is_integer(), other.dtype().is_integer()) {
-        (true, true) => {
-            let cast1 = s.cast(&DataType::Int64)?;
-            let mut iter1 = cast1.i64()?.into_iter();
-
-            match other.strict_cast(&DataType::UInt32) {
-                Ok(casted) => {
-                    let mut iter2 = casted.u32()?.into_iter();
-
-                    let res = if s.len() == 1 {
-                        let v1 = iter1.next().unwrap();
-                        iter2
-                            .map(|v2| v1.and_then(|left| v2.map(|right| left.pow(right))))
-                            .collect()
-                    } else if other.len() == 1 {
-                        let v2 = iter2.next().unwrap();
-                        iter1
-                            .map(|v1| v1.and_then(|left| v2.map(|right| left.pow(right))))
-                            .collect()
-                    } else {
-                        iter1
-                            .zip(iter2)
-                            .map(|(v1, v2)| v1.and_then(|left| v2.map(|right| left.pow(right))))
-                            .collect()
-                    };
-
-                    Ok(ExSeries::new(res))
-                }
-                Err(_) => Err(ExplorerError::Other(
-                    "negative exponent with an integer base".into(),
-                )),
-            }
-        }
-        (_, _) => {
-            let cast1 = s.cast(&DataType::Float64)?;
-            let cast2 = other.cast(&DataType::Float64)?;
-            let mut iter1 = cast1.f64()?.into_iter();
-            let mut iter2 = cast2.f64()?.into_iter();
-
-            let res = if s.len() == 1 {
-                let v1 = iter1.next().unwrap();
-                iter2
-                    .map(|v2| v1.and_then(|left| v2.map(|right| left.powf(right))))
-                    .collect()
-            } else if other.len() == 1 {
-                let v2 = iter2.next().unwrap();
-                iter1
-                    .map(|v1| v1.and_then(|left| v2.map(|right| left.powf(right))))
-                    .collect()
-            } else {
-                iter1
-                    .zip(iter2)
-                    .map(|(v1, v2)| v1.and_then(|left| v2.map(|right| left.powf(right))))
-                    .collect()
-            };
+pub fn s_pow(
+    base_exseries: ExSeries,
+    base_exdtype: ExSeriesDtype,
+    exponent_exseries: ExSeries,
+    exponent_exdtype: ExSeriesDtype,
```

Comment:
Is there a way to pull the dtype off an `ExSeries`? I couldn't figure it out, so I just passed them in as extra args. 😅

Reply:
> Is there a way to pull the dtype off an ExSeries?
`s.dtype()` or `e.dtype()`

https://github.com/elixir-explorer/explorer/blob/f06700ccb98f2a55f3a64b114d5be1b673185f33/native/explorer/src/datatypes.rs#L28-L41

https://docs.rs/polars/latest/polars/series/trait.SeriesTrait.html#method.dtype

Reply:
Thank you!

* [66b45ce](https://github.com/elixir-explorer/explorer/pull/821/commits/66b45ce4228252efdd02a55fba775816ed261ccd)
* [ed5338f](https://github.com/elixir-explorer/explorer/pull/821/commits/ed5338f6085507e7e4780dd31026a19dedb27f03)